DateTimeCalendar/Date.java

```diff
@@ -0,0 +1,322 @@
+    }
+
+    public Date(int input){
+        setDayspan(input);
+    }
+// End Constructors
+
+// Methods
+    public Date subtract(int years, int months, int days){
+        int y = this.year;
+        int m = this.month.ordinal() + 1;
+        int d = this.day;
+
+        if (years >= 0 && months >= 0 && days >= 0) {
+            y -= years;
+
+            if (m - months < 1){
+                while (m - months < 1){
+                    y--;
+                    months -= 12;
+                }
+            }
+            m -= months;
+
+            if (d - days < 1){
+                while(d - days < 1) {
+                    if (m - 1 < 1){
+                        y--;
+                        m = 12;
+                    } else {
+                        m--;
+                    }
+
+                    if (m == 2) {
+                        if (isLeapYear(y)) {
+                            days -= Date.Month.values()[m - 1].days + 1;
+                        }
+                    } 
+
+                    days -= Date.Month.values()[m - 1].days;
+                }
+            } 
+            d -= days;
+        } else {
+            throw new IllegalArgumentException("Invalid input");
+        }
+        
+        //Returning new object is a design decision. Unsure if new object or reference would be needed, to play it safe returning new obj. Example: returning vacation days might need to be a new hash.
+        return new Date(y, Date.Month.values()[m - 1], d);
+    }
+
+    public int diff(Date input){
+        //return in value of days
+        int payload = 0;
+        int currentDays = (int)convToDays(this.year, this.month.ordinal() + 1, this.day);
+        int inputDays = (int)convToDays(input.year, input.month.ordinal() + 1, input.day);
+
+        if (currentDays > inputDays){
+            payload = currentDays - inputDays;
+        } else {
+            payload = inputDays - currentDays;
+        }
+        return payload;
+    }
+
+    public Date add(int years, int months, int days){
+
+        if (years >= 0 && months >= 0 && days >= 0){
+            years += this.year;
+            months += this.month.ordinal() + 1;
+            days += this.day;
+
+            //handle Months
+            if (months > 12) {
+                years += months / 12;
+                months = months % 12;
+            }
+            //set months back to 0-11
+            months--;
+
+            while (days > 27){
+                int currentMonthDayCount = Date.Month.values()[months].days;
+                if (months == 1 && Date.isLeapYear(years)){
+                    currentMonthDayCount++;
+                }
+
+                if (days > currentMonthDayCount){
+                    if (days > DAYS_IN_REG_YEAR){
+                        if (isLeapYear(years)){
+                            days -= DAYS_IN_LEAP_YEAR;
+                        } else {
+                            days -= DAYS_IN_REG_YEAR;
+                        }
```

Comment:
enter the ternary operator:
```java
days -= isLeapYear(years) ? DAYS_IN_LEAP_YEAR : DAYS_IN_REG_YEAR;
```

Reply:
Ohh I love this and want to know more about this type of function, would like to discuss at next meeting.

Reply:
This is changed and will be in next push